### DISCUSSION TITLE
Can Weblate escape single quotes automatically in Java properties files?

### DISCUSSION THREAD

**Discussion by peter-catalin** [WeblateOrg/weblate#10724] [Q&A] [answered]
Hello there,

I'm wondering if Weblate has a setting or add-on that can automatically escape single quotes in Java files. This would alleviate the need for translators to manually adjust the escaping, particularly in French translations.

For example, in the screenshot below, the translation requires manually adding escaped quotes (two consecutive single quotes):

<img width="1425" alt="Screenshot 2024-01-04 at 10 36 34" src="https://github.com/WeblateOrg/weblate/assets/36492048/35d51bc3-7428-4a26-87fd-484411c06b67">

Ideally, translators should be able to enter "Industries d'extraction" instead of "Industries d''extraction". Weblate would then automatically detect that it's dealing with a Java properties file and apply the necessary escaping.

This can be really confusing for translators when dealing with multiple types of files. This is not an issue when dealing with JSON translation files in JavaScript.

Is there a configuration option in Weblate to enable this kind of automatic escaping?

Thank you!

**Comment by nijel** [comment, marked as answer]
That's what GWT properties use, so all you need to do is configure Weblate to use that format (https://docs.weblate.org/en/latest/formats/gwt.html).

**Reply by peter-catalin** [reply]
Works perfectly! Thank you very much 🙏
